lang/llvm-17/Portfile

```diff
@@ -224,6 +224,11 @@ if {${os.platform} eq "darwin" && ${os.major} <= 16} {
     patchfiles-append 0042-mbstate_t-not-defined.patch
 }
 
+if {${os.platform} eq "darwin" && ${os.major} >= 23} {
```

Comment:
Wouldn't this problem affect any version of any OS if building with clang 18 or later?

Reply:
If you’re bootstraping with clang-18 or later, yes.

Does this ever bootstrap with a clang that doesn’t come from Xcode or CLT, though?

Reply:
In practise this will only ever be a problem on OSes that can use Xcode 16, so Darwin23 or newer.

( I suspect its probably safe to apply unconditionally on darwin version, given clang-18 and clang-19 build just fine on the older OS, and already have this, but I would prefer to limit it until proven we need it ).

Reply:
@markmentovai Could you please update this with the same fixes for the older clang versions where it also works ?

Reply:
https://github.com/macports/macports-ports/pull/25912#discussion_r1774072183:
> ( I suspect its probably safe to apply unconditionally on darwin version, given clang-18 and clang-19 build just fine on the older OS, and already have this, but I would prefer to limit it until proven we need it ).

I actually had a preference for doing it unconditionally ([mentioned here](https://trac.macports.org/ticket/70779#comment:14), “not incorrect in any circumstance”), but noticed that you had a preference for conditionalizing the [gcc-14/x86_64 fix](https://trac.macports.org/ticket/70866#comment:14), which is why I included it here too.

https://github.com/macports/macports-ports/pull/25912#discussion_r1774073970:
> @markmentovai Could you please update this with the same fixes for the older clang versions where it also works ?

Yes, will do after the builds complete.

Reply:
Indeed, I normally prefer to limit patches to only when they are needed, so keep the source as close to vanilla upstream as we can, but there are pros and cons to this. When the patch is anyway something upstream have applied themselves in newer releases (and not a workaround we concocted ourselves) such as this case applying it unconditionally is probably fine. So in this case I will leave it to you to decide…

Reply:
> Could you please update this with the same fixes for the older clang versions where it also works ?

It would be great if the changes for other clang versions could happen in separate PRs each merged at least a day after the previous one because building any single version of clang takes the build machines a long time and delays other scheduled builds.

> I normally prefer to limit patches to only when they are needed

And my point is that it is needed when building with clang 18 on any OS version so it should not be conditional on the OS version. For example, I am on macOS 12 and I frequently instruct MacPorts to use clang 18 to build a port in order to reproduce and fix issues encountered by users using newer versions of Xcode.

Reply:
Ah, in that case yes lets apply unconditionally.

and yes, I almost suggested separate MRs for each clang in order for the ci builds to not timeout, but its just as important for the buildbots so @markmentovai indeed please open separate MRs for each clang.

Reply:
Thanks, @ryandesign and @cjones051073.

I’ve made this patch apply without condition.

I’ll send a separate pull request for each clang version that I restore compatibility for, so that you can choose the best timing to merge them.